core/dataset_manager.py

```diff
@@ -152,69 +160,68 @@ def _create_dataset_mappings(self, mapping_filepath):
         with open(mapping_filepath, "w") as f:
             f.write(yaml.dump(mappings))
 
-    def check_key_length(self, key):
+    def _generate_ed_directory(self):
         """
-        Keys for datasets can only be at most 30 characters long.
+        This is a helper method that encompasses the logic for creating the
+        ED Directory. Given the directory example in the class docstring, 
+        an example of an ED Directory can look like this:
+
+        {
+            ...
+            <dataset1_uuid> : (<dataset1 sample json>, <dataset1 metadata json>),
+            <dataset2_uuid> : (<dataset2 sample json>, <dataset2 metadata json>)
+            ...
+        }
         """
-        if len(key) > 30:
-            raise InvalidKeyError(key)
+        ed_directory = {}
+        mappings = self._mappings
+        filepath = os.path.join(self._raw_filepath, 'datasets.yaml')
+        reverse_mappings = {v: k for k, v in mappings.items()}
 
-    def post_dataset_with_md(self, name):
+        for folder in os.listdir(self._raw_filepath):
+            folder_path = os.path.join(os.path.abspath(self._raw_filepath), folder)
+            if not os.path.isdir(folder_path): continue
+            encoding = reverse_mappings[folder]
+            files = list(os.listdir(folder_path))
+            assert len(files) == 1, \
+                "We only support one file per dataset folder!"
+            file_name = files[0]
+            if not file_name.endswith(".csv"): continue
+            filepath = os.path.join(folder_path, file_name)
+            dataset = pd.read_csv(filepath)
+            sample = dataset.sample(frac=0.1)
```

Comment:
Filesize of sample data shouldn't scale sublinearly to size of real data, it should scale log or preferrably constant. You don't need to fix this in this PR.

Reply:
Noted, thanks